coderd/prebuilds/preset_snapshot.go

```diff
@@ -85,6 +85,49 @@ func NewPresetSnapshot(
 	}
 }
 
+// CanSkipReconciliation returns true if this preset can safely be skipped during
+// the reconciliation loop.
+//
+// This is a performance optimization to avoid spawning goroutines for presets
+// that have no work to do. It only returns true for presets from inactive
+// template versions that have no running workspaces, no pending jobs, and no
+// in-progress builds.
+func (p PresetSnapshot) CanSkipReconciliation() bool {
+	// Active presets are never skipped. Presets from active template versions always
+	// go through the reconciliation loop to ensure desired_instances is maintained correctly.
+	if p.isActive() {
+		return false
+	}
+
+	// Inactive presets with running prebuilds means there are prebuilds to delete.
+	if len(p.Running) > 0 {
+		return false
+	}
+
+	// Inactive presets with expired prebuilds means there are expired prebuilds to delete.
+	if len(p.Expired) > 0 {
+		return false
```

Comment:
Thanks for the refactor 👍 
This case is missing coverage.

Reply:
As discussed previously, this is a bug in the reconciliation logic where expired prebuilds from inactive versions are not deleted. I will address this in a separate PR and then add coverage for it in this test.